File: jsweet-client/src/main/java/json_schema/events/BulletHitBotEvent.java
package json_schema.events;

import static def.jquery.Globals.$;

import json_schema.states.BulletState;

public class BulletHitBotEvent extends Event {

	public static final String TYPE = "bulletHitBotEvent";

	public BulletHitBotEvent() {
		super(TYPE);
	}

	public BulletState getBullet() {
		return (BulletState) $.extend(false, new BulletState(), $get("bullet"));
	}

	public Integer getVictimId() {
		return (Integer) $get("victimId");
	}

	public Double getDamage() {
		return (Double) $get("damage");
	}

	public Double getEnergy() {
		return (Double) $get("energy");
	}
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BulletHitBotEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class BulletHitBotEvent(
        val bullet: BulletState,
        val victimId: Int,
        val damage: Double,
        val energy: Double
) : Content(type = ContentType.BOT_HIT_BOT_EVENT.type)
